src/main/java/codeu/controller/AdminServlet.java

```diff
@@ -0,0 +1,198 @@
+
+/** Servlet class responsible for the admin page. */
+public class AdminServlet extends HttpServlet {
+
+	 /** Store class that gives access to Conversations. */
+	  private ConversationStore conversationStore;
+
+	  /** Store class that gives access to Messages. */
+	  private MessageStore messageStore;
+
+	  /** Store class that gives access to Users. */
+	  private UserStore userStore;
+
+	  /** Set up state for handling chat requests. */
+	  @Override
+	  public void init() throws ServletException {
+	    super.init();
+	    setConversationStore(ConversationStore.getInstance());
+	    setMessageStore(MessageStore.getInstance());
+	    setUserStore(UserStore.getInstance());
+	  }
+
+	  /**
+	   * Sets the ConversationStore used by this servlet. This function provides a common setup method
+	   * for use by the test framework or the servlet's init() function.
+	   */
+	  void setConversationStore(ConversationStore conversationStore) {
+	    this.conversationStore = conversationStore;
+	  }
+
+	  /**
+	   * Sets the MessageStore used by this servlet. This function provides a common setup method for
+	   * use by the test framework or the servlet's init() function.
+	   */
+	  void setMessageStore(MessageStore messageStore) {
+	    this.messageStore = messageStore;
+	  }
+
+	  /**
+	   * Sets the UserStore used by this servlet. This function provides a common setup method for use
+	   * by the test framework or the servlet's init() function.
+	   */
+	  void setUserStore(UserStore userStore) {
+	    this.userStore = userStore;
+	  }
+
+	  /**
+	   * This function fires when a user requests the /login URL. It simply forwards the request to
+	   * admin.jsp.
+	   */
+	  @Override
+	  public void doGet(HttpServletRequest request, HttpServletResponse response)
+	      throws IOException, ServletException 
+	  {
+		  List<Conversation> conversations = conversationStore.getAllConversations();
+		  request.setAttribute("conversations", conversations);
+		  
+		  int numConvos = getNumConvos(conversations);
+		  request.setAttribute("numConvos", numConvos);
+		  
+		  String activeUserName = getActiveUser(conversations);
+		  request.setAttribute("activeUserName", activeUserName);
+		  
+		  int numMessages = getNumMessages(conversations);
+		  request.setAttribute("numMessages", numMessages);
+		  
+		  int numUsers = userStore.numUsers();
+		  request.setAttribute("numUsers", numUsers);
+		  
+		  request.getRequestDispatcher("/WEB-INF/view/admin.jsp").forward(request, response);
+	  }
+	  
+	  //Calculates the most popular value in a List
+	  public UUID getPopularElement(List<UUID> a)
+	  {
+		if (a == null)
+			return null;
+		else
+		{
+			int count = 1, tempCount;
+		    UUID popular = a.get(0);
+		    UUID temp = null;
+		    for (int i = 0; i < (a.size() - 1); i++)
+		    {
+		      temp = a.get(i);
+		      tempCount = 0;
+		      for (int j = 1; j < a.size(); j++)
+		      {
+		        if (temp == a.get(j))
```

Comment:
Would add {} to this if too. It's more readable if you are consistent throughout all the statements.